error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/OptionalRules.java

```diff
@@ -220,6 +222,22 @@ T after(Optional<T> o1, Optional<T> o2) {
     }
   }
 
+  /**
+   * Prefer {@link Optional#orElseGet(Supplier)} over {@link Optional#orElse(Object)} if the given
+   * value is not a compile-time constant.
```

Comment:
I'll need to think a bit about phrasing, but likely we should call out that this rule can, in rare circumstances, change application behavior. (This can happen if `value` matches a non-pure expression.)